cpp-inference/inference_scaling_imagenet.cpp

```diff
@@ -69,34 +128,55 @@ void run_mnist(const std::string& model_name,
                       SmartRedis::MemoryLayout::nested);
     double put_tensor_end = MPI_Wtime();
     delta_t = put_tensor_end - put_tensor_start;
-    timing_file << rank << "," << "put_tensor" << ","
-                << delta_t << std::endl << std::flush;
+    put_tensor_times.push_back(delta_t);
+    //timing_file << rank << "," << "put_tensor" << ","
+    //            << delta_t << std::endl << std::flush;
 
     double run_script_start = MPI_Wtime();
     client.run_script(script_name, "pre_process_3ch", {in_key}, {script_out_key});
     double run_script_end = MPI_Wtime();
     delta_t = run_script_end - run_script_start;
-    timing_file << rank << "," << "run_script" << ","
-                << delta_t << std::endl << std::flush;
+    run_script_times.push_back(delta_t);
+    //timing_file << rank << "," << "run_script" << ","
+    //            << delta_t << std::endl << std::flush;
 
     double run_model_start = MPI_Wtime();
     client.run_model(model_name, {script_out_key}, {out_key});
     double run_model_end = MPI_Wtime();
     delta_t = run_model_end - run_model_start;
-    timing_file << rank << "," << "run_model" << ","
-                << delta_t << std::endl << std::flush;
+    run_model_times.push_back(delta_t);
+    //timing_file << rank << "," << "run_model" << ","
+    //            << delta_t << std::endl << std::flush;
 
   double unpack_tensor_start = MPI_Wtime();
   client.unpack_tensor(out_key, result, {1,1000},
                        SmartRedis::TensorType::flt,
                        SmartRedis::MemoryLayout::nested);
   double unpack_tensor_end = MPI_Wtime();
   delta_t = unpack_tensor_end - unpack_tensor_start;
-  timing_file << rank << "," << "unpack_tensor" << ","
-              << delta_t << std::endl << std::flush;
+  unpack_tensor_times.push_back(delta_t);
+  //timing_file << rank << "," << "unpack_tensor" << ","
+  //            << delta_t << std::endl << std::flush;
   }
   double loop_end = MPI_Wtime();
   delta_t = loop_end - loop_start;
+
+  // write times to file
+  for (int i=0; i<100; i++) {
```

Comment:
Maybe we define a constant for the number of trials now that we use the constant (e.g. ``100``) in two loops?